std-bits/table/src/main/java/org/enso/table/data/column/operation/cast/ToDateStorageConverter.java

```diff
@@ -0,0 +1,58 @@
+package org.enso.table.data.column.operation.cast;
+
+import org.enso.table.data.column.builder.object.DateBuilder;
+import org.enso.table.data.column.storage.DateStorage;
+import org.enso.table.data.column.storage.DateTimeStorage;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.column.storage.type.AnyObjectType;
+
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+
+public class ToDateStorageConverter implements StorageConverter<LocalDate> {
+  public ToDateStorageConverter() {
+  }
+
+  public Storage<LocalDate> cast(Storage<?> storage, CastProblemBuilder problemBuilder) {
+    if (storage instanceof DateStorage dateStorage) {
+      return dateStorage;
+    } else if (storage instanceof DateTimeStorage dateTimeStorage) {
+      return convertDateTimeStorage(dateTimeStorage);
+    }else if (storage.getType() instanceof AnyObjectType) {
+      return castFromMixed(storage, problemBuilder);
+    } else {
+      throw new IllegalStateException("No known strategy for casting storage " + storage + " to Date.");
+    }
```

Comment:
Not easily, because the first 2 branches switch on `storage` but the third one on `storage.getType` - there are 2 storages (and we may have more in the future) that may return `AnyObjectType` - the `ObjectStorage` and the `MixedStorageFacade`.